src/schema/schema-generator.ts

```diff
@@ -373,15 +373,25 @@ export abstract class SchemaGenerator {
       if (handler.resolverClassMetadata && handler.resolverClassMetadata.isAbstract) {
         return fields;
       }
+      const pubSubIterator = (payload: any, args: any, context: any, info: any) => {
+        let topics: string | string[];
+        if (typeof handler.topics === "function") {
```

Comment:
Due to performance reasons, I would prefer do this check outside the `pubSubIterator` function - we should have a simple function for topics that are constants strings and the one that just call the topics function and `asyncIterator` with the result, but no runtime `typeof` check on every call of `pubSubIterator` function

Reply:
```ts
let pubSubIterator: ResolverFn;
if (typeof handler.topics === "function") {
  const getTopics = handler.topics;
  pubSubIterator = (payload, args, context, info) => {
    const resolverTopicData: ResolverTopicData = { payload, args, context, info };
    const topics = getTopics(resolverTopicData);
    return pubSub.asyncIterator(topics);
  };
} else {
  const topics = handler.topics;
  pubSubIterator = () => pubSub.asyncIterator(topics);
}
```